src/server.ts

```diff
@@ -20,29 +25,59 @@ export class Server {
   public startServer() {
     return this.app.listen(process.env.PORT || 8080, (err: any) => {
       if (err) return console.error(err);
-      else if (process.env.NODE_ENV === 'development') return console.log('server ready');
-
+      else if (process.env.NODE_ENV === 'development') return console.log("\x1b[35m\x1b[47m", "server ready", "\x1b[0m");
     });
   }
 
   private checkRequest(url: string, apiKey: string, res: Express.Response) {
     return new Promise((resolve, reject) => {
-      console.log(process.env.API_KEY);
-      if (!url || !apiKey) {
+      if (!url || (!!process.env.API_KEY && !apiKey)) {
         reject(res.status(400).send({
-          success: false,
           message: "Bad call",
         }));
       } else if (!!process.env.API_KEY && apiKey != process.env.API_KEY) {
         reject(res.status(401).send({
-          success: false,
           message: "Bad API key",
         }));
       }
       resolve();
     })
   }
 
+  private readPDFParams(queyParams: any) {
+    Object.keys(queyParams).forEach(param => {
+      if (!this.pdfParams.margin) this.pdfParams.margin = {};
+      switch(param) {
+        case 'format':
+          this.pdfParams.format = queyParams.format;
+          break;
+        case 'scale':
+          this.pdfParams.scale = +queyParams.scale;
+          break;
+        case 'marginTop':
+          this.pdfParams.margin.top = queyParams.marginTop;
+          break;
+        case 'marginLeft':
+          this.pdfParams.margin.left = queyParams.marginLeft;
+          break;
+        case 'marginBottom':
+          this.pdfParams.margin.bottom = queyParams.marginBottom;
+          break;
+        case 'marginRight':
+          this.pdfParams.margin.right = queyParams.marginRight;
+          break;
+        case 'width':
+          this.pdfParams.width = queyParams.width;
+          break;
+        case 'height':
+          this.pdfParams.height = queyParams.height;
+          break;
+        case 'tilte':
+          this.pdfParams.title = queyParams.title;
+      }
+    });
+  }
+
```

Comment:
wow this is ugly. You don't need that much code to do this. Also, be aware that adding parameters has a negative impact on components coupling